src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/RazorPage/RazorPageScaffolderBase.cs

```diff
@@ -140,6 +176,33 @@ protected RazorPageGeneratorTemplateModel GetRazorPageViewGeneratorTemplateModel
             return templateModel;
         }
 
+        protected bool RazorPagesFolderExists(string relativeFolderPath, string applicationBasePath)
```

Comment:
This method is identical to to ViewScaffolderBase.ViewsFolderExists except for the value of the `const string viewsFolderName`. I'd suggest moving the method to CommonGeneratorBase and passing in the viewsFolderName as another param.